sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScalerProvider.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Azure.WebJobs.Extensions.ServiceBus.Config;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.ServiceBus;
+using Microsoft.Azure.WebJobs.ServiceBus.Listeners;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Listeners
+{
+    internal class ServiceBusScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly IConfiguration _configuration;
+        private AzureComponentFactory _defaultAzureComponentFactory;
+        private readonly MessagingProvider _messagingProvider;
+        private readonly IOptions<ServiceBusOptions> _options;
+        private readonly AzureEventSourceLogForwarder _logForwarder;
+        private readonly TriggerMetadata _triggerMetadata;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public ServiceBusScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            _configuration = serviceProvider.GetService<IConfiguration>();
+            _defaultAzureComponentFactory = triggerMetadata.Properties[nameof(AzureComponentFactory)] as AzureComponentFactory;
+            _messagingProvider = serviceProvider.GetService<MessagingProvider>();
+            _triggerMetadata = triggerMetadata;
+            _logForwarder = serviceProvider.GetService<AzureEventSourceLogForwarder>();
+            _options = serviceProvider.GetService<IOptions<ServiceBusOptions>>();
+            _loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+        }
+
+        public IScaleMonitor GetMonitor()
+        {
+            ServiceBusMetadata serviceBusMetadata = JsonConvert.DeserializeObject<ServiceBusMetadata>(_triggerMetadata.Metadata.ToString());
```

Comment:
Can avoid the double serialization/deserialization using techniques like https://stackoverflow.com/questions/30220328/how-can-i-populate-an-existing-object-from-a-jtoken-using-newtonsoft-json

Reply:
Are you proposing to inherit `ServiceBusMetadata` from `TriggerMetadata` here? We need a new `ctor` for `TriggerMetadata(TriggerMetadata triggerMetadata)` what is also not very clean.